### PR TITLE
Create konbini gateway for Komoju

### PR DESCRIPTION
**Pull request by camelmasa**
I added konbini gateway for [Komoju](https://www.degica.com/products/komoju/).

## TODO

(We are need to waiting until merge `komoju/active_merchant` to `Shopify/active_merchant`)

- [ ] Update activemerchant gem

### CODE REVIEW

**Comment by rramsden** [comment]
For the payment logos there's a set of images on the bottom of the docs page https://docs.komoju.com/#logos

**Comment by camelmasa** [comment]
@rramsden Thanks. I'll replace those images :ok_hand:

File: app/controllers/spree/checkout_controller_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  def permitted_source_attributes
+    super.push(permittd_komoju_konbini_attributes)
+    super.push(permittd_komoju_banktransfer_attributes)
+    super.push(permittd_komoju_pay_easy_attributes)
+    super.push(permittd_komoju_web_money_attributes)
+  end
+
+  private
+
+  def permittd_komoju_konbini_attributes
```
Comment (shioyama) [inline app/controllers/spree/checkout_controller_decorator.rb]: "permittd" -> "permitted"

Comment (camelmasa) [inline app/controllers/spree/checkout_controller_decorator.rb]: :+1:

**Comment by camelmasa** [comment]
@shioyama I created konbini gateway for komoju. Please review when you have time.
I created other gateways on [this branch](https://github.com/camelmasa/spree_komoju/tree/add-komoju-gateways).

I'll create PR for other gateway after merged this PR.

**Comment by shioyama** [comment]
Great! Reviewing now.

File: app/models/spree/komoju_gateway.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    end
+
+    def supports?(source)
+      true
+    end
+
+    # We need to change price from cents to dollar for Komoju gateway.
+    # Because, Komoju gateway supports JPY currency only.
+    def change_options_to_dollar(options)
+      %i(shipping tax subtotal discount).each { |key| options[key] = options[key] / 100.0 }
```
Comment (shioyama) [inline app/models/spree/komoju_gateway.rb]: I'm not exactly clear why this is needed, can you explain later?

Comment (shioyama) [inline app/models/spree/komoju_gateway.rb]: Ref: https://github.com/spree/spree/blob/master/core/app/models/spree/payment/gateway_options.rb

Comment (camelmasa) [inline app/models/spree/komoju_gateway.rb]: Yes. I'll add [the link](https://github.com/spree/spree/blob/master/core/app/models/spree/payment/gateway_options.rb) to here.

**Comment by shioyama** [comment]
@camelmasa this looks good, one question: do we want to show response details in the complete page?

File: app/models/spree/gateway/komoju_konbini.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+module Spree
+  class Gateway::KomojuKonbini < KomojuGateway
+    def authorize(money, source, options)
+      details = {
+        type:  "konbini",
+        store: source.convenience,
+        phone: options[:billing_address][:phone]
+      }
+      options = change_options_to_dollar(options) if options[:currency] == "JPY"
```
Comment (shioyama) [inline app/models/spree/gateway/komoju_konbini.rb]: I wonder: should we perhaps just block any payment that is not in "JPY" for now? Or raise an error.

**Comment by camelmasa** [comment]
@shioyama Thank you for reviewing :octocat: 

Yes, we want to show response details. But we need to change `config.eager_load` to `true` if we show that.

Then, I'll check about that.

**Comment by shioyama** [comment]
LGTM :+1: great work

File: app/models/spree/gateway/komoju_konbini.rb (hunk 3 above)
Comment (camelmasa) [inline app/models/spree/gateway/komoju_konbini.rb]: I think no problem. Komoju can captures only JPY.
Spree will receives error response from Komoju, then spree will saves the error log.

**Pull request closed without merging**